### ISSUE TITLE
Login facebook does not work on a specific device (Xiaomi redmi note 9)

### ISSUE THREAD

**Issue by georgeMorales** [expo/expo#10813]
Expo sdk 39
“expo-facebook”: “~9.0.0”
Platform: Android
Managed workflow

I have an application that works with the facebook login, in the emulator it works fine, on my Xiaomi MI9 device it works fine, on other devices such as huawei, samsung that I have tested it works fine, however with a Xiaomi redmi note 9 device an error appears with facebook login and I don’t know how to fix it since it works fine on the other devices.
The error appears in expo client and in the standalone application on that Xiaomi redmi note 9 device.
This is the error it throws me when I press the button:
An error occurred while trying to log in to Facebook

node_modules/react-native/Libraries/BatchedBridge/NativeModules.js:103:50 in promiseMethodWrapper
node_modules/@unimodules/react-native-adapter/build/NativeModulesProxy.native.js:15:23 in moduleName.methodInfo.name
http://packager.vn-cvh.jorgem.le-firestore-1-sdk39.exp.direct/node_modules/expo/AppEntry.bundle?platform=android&dev=true&minify=false&hot=false:299953:103 in logInWithReadPermissionsAsync$
node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
node_modules/regenerator-runtime/runtime.js:293:29 in invoke
node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
node_modules/regenerator-runtime/runtime.js:154:27 in invoke
node_modules/regenerator-runtime/runtime.js:189:16 in PromiseImpl$argument_0
node_modules/react-native/node_modules/promise/setimmediate/core.js:45:6 in tryCallTwo
node_modules/react-native/node_modules/promise/setimmediate/core.js:200:22 in doResolve
node_modules/react-native/node_modules/promise/setimmediate/core.js:66:11 in Promise
node_modules/regenerator-runtime/runtime.js:188:15 in callInvokeWithMethodAndArg
node_modules/regenerator-runtime/runtime.js:211:38 in enqueue
node_modules/regenerator-runtime/runtime.js:238:8 in exports.async
features/eventFeatures/Events/Events.js:571:12 in componentDidUpdate
app/common/card/LogoCard.js:18:37 in LogoCard
node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
node_modules/regenerator-runtime/runtime.js:293:29 in invoke
node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
node_modules/regenerator-runtime/runtime.js:154:27 in invoke
node_modules/regenerator-runtime/runtime.js:164:18 in PromiseImpl.resolve.then$argument_0
node_modules/react-native/node_modules/promise/setimmediate/core.js:37:13 in tryCallOne
node_modules/react-native/node_modules/promise/setimmediate/core.js:123:24 in setImmediate$argument_0
node_modules/react-native/Libraries/Core/Timers/JSTimers.js:130:14 in _callTimer
node_modules/react-native/Libraries/Core/Timers/JSTimers.js:181:14 in _callImmediatesPass
node_modules/react-native/Libraries/Core/Timers/JSTimers.js:441:30 in callImmediates
node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:387:6 in __callImmediates
node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:135:6 in __guard$argument_0
node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:364:10 in __guard
node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:134:4 in flushedQueue
[native code]:null in flushedQueue
[native code]:null in invokeCallbackAndReturnFlushedQueue

**Comment by JesseRWeigel** [comment]
I am getting the same error with a Pixel 3 XL.

**Comment by leianie** [comment]
I'm also getting the same error on several devices using sdk 39

**Comment by fabioglimb** [comment]
I've encountered this error too, i managed to log the error on Sentry and it says this: 

Encountered an exception while calling native method: Exception occurred while executing exported method logInWithReadPermissionsAsync on module ExponentFacebook: The SDK has not been initialized, make sure to call FacebookSdk.sdkInitialize() first.

I've tried on multiple android devices and tried to put the await Facebook.initializeAsync({}) in different places inside my App.tsx, it's not working.

That occures only on Android, IOS is working fine, i've tested on Huawei P8 Lite, One plus 5T and on Xiaomi redminote 9s.

I am on Expo SDK 39.0.5 and expo-facebook 9.1.0, if someone managed to make it work or can help me with a stable version I would be really grateful.

**Comment by nncl** [comment]
The same is happening to me on Samsung S10:

```json
"expo": "39.0.5",
"expo-facebook": "9.0.0"
```

Does anyone know anything about this?

**Comment by georgeMorales** [comment]
Nothing :(

**Comment by PieroSantistebanQ** [comment]
Same issue Nova 5T

**Comment by PieroSantistebanQ** [comment]
I fixed, build first time the app and generate the hashes.
SDK 40

**Comment by fabioglimb** [comment]
@PieroSantistebanQ how did you managed to resolve it? I've upgraded every module, moved to SDK 40 and last version of expo-cli and still not working on android devices..

**Comment by PieroSantistebanQ** [comment]
> @PieroSantistebanQ how did you managed to resolve it? I've upgraded every module, moved to SDK 40 and last version of expo-cli and still not working on android devices..

First execute expo build:android and generate the hashes

**Comment by fabioglimb** [comment]
@PieroSantistebanQ already done it, did you set "facebookAutoInitEnabled": true in your app.json or did you init the sdk manually with Facebook.initializeAsync ?

**Comment by PieroSantistebanQ** [comment]
> @PieroSantistebanQ already done it, did you set "facebookAutoInitEnabled": true in your app.json or did you init the sdk manually with Facebook.initializeAsync ?
> 

With Facebook.initializeAsync

**Comment by fabioglimb** [comment]
@PieroSantistebanQ I've tried even like this and nothing seems to work, only on android and still getting that the sdk has not been initialised. Could you provide a snippet of the inizialization and the modules in your package json? I'll really appreciate that

**Comment by PieroSantistebanQ** [comment]
> @PieroSantistebanQ I've tried even like this and nothing seems to work, only on android and still getting that the sdk has not been initialised. Could you provide a snippet of the inizialization and the modules in your package json? I'll really appreciate that

await Facebook.initializeAsync({ appId: '000000000', appName: 'AAAAA' })
const {
			type,
			token,
			expires,
			permissions,
			declinedPermissions
		} = await Facebook.logInWithReadPermissionsAsync({
			permissions: [ 'public_profile', 'email' ]
		})

"expo": "^40.0.0",
"expo-auth-session": "~3.0.0",
"expo-facebook": "~9.1.0",

**Comment by fabioglimb** [comment]
@PieroSantistebanQ thank you, now it's working. And for everyone that is encountering this error, you MUST call the await await Facebook.initializeAsync() passing appId and appName before calling await Facebook.logInWithReadPermissionsAsync, otherwise it will say that the sdk has not been initialized yet. Hope it will be fixed so that the "autoInitEnabled: true"
 prop for the app.json will make it work.

**Comment by Samuel-Sorial** [comment]
> @PieroSantistebanQ thank you, now it's working. And for everyone that is encountering this error, you MUST call the await await Facebook.initializeAsync() passing appId and appName before calling await Facebook.logInWithReadPermissionsAsync, otherwise it will say that the sdk has not been initialized yet. Hope it will be fixed so that the "autoInitEnabled: true"
> prop for the app.json will make it work.

I'm encountring the same problem even after calling Facebook.initializeAsync(), here's my code:
const handleFacebook = async (event) => {
  try {
    await Facebook.initializeAsync({appId: 'myappid'});
    const {
      type,
      token,
      expirationDate,
      permissions,
      declinedPermissions,
    } = await Facebook.logInWithReadPermissionsAsync({
      permissions: ['public_profile'],
    });
    console.log(type);
    if (type === 'success') {
      // Get the user's name using Facebook's Graph API
      const response = await fetch(
        `https://graph.facebook.com/me?access_token=${token}`
      );
      Alert.alert('Logged in!', `Hi ${(await response.json()).name}!`);
    } else {
      // type === 'cancel'
    }
  } catch (error) {
    console.error(error);
    alert(`Facebook Login Error: ${error.message}`);
  }
};

Any help?

**Comment by Cyrusrome22** [comment]
Got the same error too.

**Comment by valentingolchev** [comment]
@Cyrusrome22 and @Samuel-Sorial make sure to pass both `appId` and 'appName'. I had this issue and it was fixed as mentioned by @fabioglimb only after adding both properties to the call to 'Facebook.initializeAsync({ appId: "aa", appName: "bb" })'.

**Comment by nkalinov** [comment]
+1 I am experiencing the same issue on some devices. So far those are:
- SM-A505FN
- SM-A405FN
- SM-A307GN

**Comment by valentingolchev** [comment]
I solved my issue by adding the hash for the "App signing key certificate" to the Facebook developer setup. Before I was using the "Upload key certificate". It was working fine with the web-flow, but failing when the FB app was installed and used for sign in.

**Comment by chengqing97** [comment]
Check this out, this solved my issue.
https://forums.expo.dev/t/facebook-login-not-working-in-standalone-android-app/39756
